### PR TITLE
IP interface loopback action HLD

### PR DESCRIPTION
**Pull request by liorghub**
This HLD document is describing IP interface loopback action feature.

Related PRs:

| PR title                                                                                                                        | State                                                                                                               | Status                              | Dependencies                                         | Unitest               | Owner                                                                                                              |
| --------------------------------------------------------------------------------------------------------------------------------|-------------------------------------------------------------------------------------------------------------------- | ------------------------------------| ---------------------------------------------------- | ----------------------| ----------|
| [[yang] Add IP interface loopback action related content to YANG models](https://github.com/Azure/sonic-buildimage/pull/11012) | GitHub issue/pull request detail  | open                                | None                                                 | Done, part of the PR  | @liorghub |
| [[utilities] Add support for IP interface loopback action](https://github.com/Azure/sonic-utilities/pull/2192)                  | GitHub issue/pull request detail    | open                                | None                                                 | Done, part of the PR  | @liorghub |
| [[swss] Add support for IP interface loopback action](https://github.com/Azure/sonic-swss/pull/2307)                            | GitHub issue/pull request detail         | open                                | None                                                 | Done, part of the PR  | @liorghub |

### CODE REVIEW

File: doc/ip-interface/loopback-action/ip-interface-loopback-action-design.md [hunk 1]
```diff
@@ -0,0 +1,359 @@
...
+3. Check table INTF_TABLE in App DB
+4. Check table SAI_OBJECT_TYPE_ROUTER_INTERFACE in ASIC DB
+5. Create port-channel IP interface and repeat steps 2 to 4.
+6. Create interface VLAN and repeat steps 2 to 4.
+7. Create subinterface and repeat steps 2 to 4.
+
+## 3.2 System tests
+Add the following test in sonic-mgmt.
+1. Create IP interface Ethernet.
+2. Loopback action not configured, send traffic, verify lopbacked packets are forwarded (SAI default).
```
Comment (prsunny) [inline doc/ip-interface/loopback-action/ip-interface-loopback-action-design.md:352]: loopback -> typo. Please fix other places as well

File: doc/ip-interface/loopback-action/ip-interface-loopback-action-design.md [hunk 2]
```diff
@@ -0,0 +1,359 @@
...
+3. Loopback action can be set to "drop" or "forward". 
+4. CLI will reject loopback action setting on a non IP interface. Interface is considered as an IP interface if it has entry in:
+    1. VLAN_INTERFACE table for interface vlan (e.g. VLAN_INTERFACE|Vlan100)
+    2. INTERFACE table for interface Ethernet (e.g. INTERFACE|Ethernet232)
+    3. PORTCHANNEL_INTERFACE table for interface port-channel (e.g. PORTCHANNEL_INTERFACE|PortChannel1)    
+    4. VLAN_SUB_INTERFACE table for subinterface (e.g. VLAN_SUB_INTERFACE|Ethernet0.10)
+
+5. The configured loopback action can be viewed by show command in CLI.
+6. When SONiC configuration of loopback action is missing the system will behave as it is today, based on SAI behavioural model, forward the loopbacked packets.
+
```
Comment (prsunny) [inline doc/ip-interface/loopback-action/ip-interface-loopback-action-design.md:77]: Please add a section to specify - no change to kernel

File: doc/ip-interface/loopback-action/ip-interface-loopback-action-design.md (hunk 1 above)
Comment (liorghub) [reply doc/ip-interface/loopback-action/ip-interface-loopback-action-design.md]: Fixed in all places.

**Review by prsunny — APPROVED** [review]

**Comment by prsunny** [comment]
Please also update the sonic-mgmt PR in description when ready

**Comment by liorghub** [comment]
> Please also update the sonic-mgmt PR in description when ready

Done.

**Pull request merged**